### PR TITLE
03 date tasks

### PR DESCRIPTION
**Pull request by Saik1s**

### CODE REVIEW

File: task/03-date-tasks.js [hunk 1]
```diff
@@ -56,7 +56,11 @@ function parseDataFromIso8601(value) {
  *    Date(2015,1,1)    => false
  */
 function isLeapYear(date) {
-   throw new Error('Not implemented');
+   if (date.getUTCFullYear()%4 === 0){
```
Comment (Geass1000) [inline task/03-date-tasks.js]: Where are your variables :)?

File: task/03-date-tasks.js [hunk 2]
```diff
@@ -56,7 +56,11 @@ function parseDataFromIso8601(value) {
...
+      return !(date.getUTCFullYear() % 400 !== 0 && date.getUTCFullYear() % 100 === 0); // if year is divisible by 4 and by 400, then it is a leap year
+   }
+   else
+      return false; // the year is not divisible by 4 then it is not a leap year
```
Comment (Geass1000) [inline task/03-date-tasks.js]: All expressions in the function are logical expressions and they return the logic `true/false` result.

File: task/03-date-tasks.js [hunk 3]
```diff
@@ -76,7 +80,7 @@ function isLeapYear(date) {
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,15,20,10,453)   => "05:20:10.453"
  */
 function timeSpanToString(startDate, endDate) {
-   throw new Error('Not implemented');
+   return new Date(endDate - startDate).toISOString().slice(11, -1);
```
Comment (Geass1000) [inline task/03-date-tasks.js]: Where are your variables :)?

Comment (Geass1000) [inline task/03-date-tasks.js]: 11 is magical number. What does it mean?

File: task/03-date-tasks.js [hunk 4]
```diff
@@ -94,7 +98,15 @@ function timeSpanToString(startDate, endDate) {
  *    Date.UTC(2016,3,5,21, 0) => Math.PI/2
  */
 function angleBetweenClockHands(date) {
-    throw new Error('Not implemented');
+   let hours = date.getUTCHours();
+   let minutes = date.getUTCMinutes();
+   let angle = 0.5 * (60* hours + minutes) - 0.5 * 12 * minutes;
```
Comment (Geass1000) [inline task/03-date-tasks.js]: All variables in the function is constant value. You can use `const` for them.

Comment (Geass1000) [inline task/03-date-tasks.js]: A lot of magic numbers. What do they mean?

File: task/03-date-tasks.js [hunk 5]
```diff
@@ -94,7 +98,15 @@ function timeSpanToString(startDate, endDate) {
...
+   if(angle > 180 && angle <= 540) 
+      return Math.abs(360 - angle) * Math.PI/180;
+   else if(angle > 540 && angle <= 900)
+      return Math.abs(720 - angle) * Math.PI/180;
+   else 
+      return angle * Math.PI/180;
```
Comment (Geass1000) [inline task/03-date-tasks.js]: You can use `if` + `return` operators without `else`.

File: task/03-date-tasks.js (hunk 4 above)
Comment (Saik1s) [reply task/03-date-tasks.js]: these are the constants in the formula

File: task/03-date-tasks.js [hunk 6]
```diff
@@ -56,7 +56,12 @@ function parseDataFromIso8601(value) {
  *    Date(2015,1,1)    => false
  */
 function isLeapYear(date) {
-   throw new Error('Not implemented');
+    const year = date.getFullYear();
+    const divisibleBy4 = year % 4 === 0;
```
Comment (Geass1000) [inline task/03-date-tasks.js]: divisible by N? What?) What is it?)) What do you mean `divisible by N`? Is it a year or month or day or something else?

File: task/03-date-tasks.js [hunk 7]
```diff
@@ -94,7 +103,25 @@ function timeSpanToString(startDate, endDate) {
...
+    const hoursRotationSpeed = 0.5;     //rotation speed of the hours hand (degrees per minute)
+    const minutesRotationSpeed  = 6;        //rotation speed of the minute hand (degrees per minute)
+
+    const hoursAngle = secondsInOneMinute * hours + minutes;
+    const minutesAngle = minutesRotationSpeed * minutes;
+
+    const angle = hoursRotationSpeed * hoursAngle - minutesAngle;
+
+    const angleLessThen180 = (angle > 180)
+        ? Math.abs(angle - 360 * Math.round(angle / 360))
```
Comment (Geass1000) [inline task/03-date-tasks.js]: `Math.abs(angle - 360 * Math.round(angle / 360))` What do you calculate in this place?

File: task/03-date-tasks.js [hunk 8]
```diff
@@ -94,7 +103,25 @@ function timeSpanToString(startDate, endDate) {
...
+    const secondsInOneMinute = 60;
+    const hoursRotationSpeed = 0.5;     //rotation speed of the hours hand (degrees per minute)
+    const minutesRotationSpeed  = 6;        //rotation speed of the minute hand (degrees per minute)
+
+    const hoursAngle = secondsInOneMinute * hours + minutes;
+    const minutesAngle = minutesRotationSpeed * minutes;
+
+    const angle = hoursRotationSpeed * hoursAngle - minutesAngle;
+
+    const angleLessThen180 = (angle > 180)
```
Comment (Geass1000) [inline task/03-date-tasks.js]: `angle Less Then 180 `  bad name of variable. It is not a boolean variable.

File: task/03-date-tasks.js (hunk 6 above)
Comment (Saik1s) [reply task/03-date-tasks.js]: everyone knows that a leap year every 4 years, with few exceptions :)

Comment (Geass1000) [reply task/03-date-tasks.js]: Not everyone. Your code must be readable for customers. Customer might not know about leap years.

**Pull request merged**